app/models/spree/order.rb

```diff
@@ -62,7 +62,11 @@ def states
     has_many :line_item_adjustments, through: :line_items, source: :adjustments
     has_many :shipment_adjustments, through: :shipments, source: :adjustments
     has_many :all_adjustments, class_name: 'Spree::Adjustment', dependent: :destroy
-    has_many :vouchers, -> { where(originator_type: 'Voucher') },
+    has_many :vouchers,
+             -> {
+               where(originator_type: 'Voucher')
+                 .order("#{Spree::Adjustment.table_name}.created_at ASC")
```

Comment:
Nice. I often wonder if it's better to keep the association simple and sort where it's displayed. I guess that there are good reasons for either way.